### DISCUSSION TITLE
stage6実行時にsize mismatch for～エラーが発生してしまう。

### DISCUSSION THREAD

**Discussion by aoi-takumi** [espnet/espnet#4472] [Q&A] [answered]
vitsのfinetuningを行った際下記エラーが発生しました。同様のdiscussionsがありましたが、理解が及ばず解決策がわからなかったので質問させてください。

ダウンロードしたモデル
espnet_model_zoo_download --unpack true --cachedir downloads kan-bayashi/jsut_vits_accent_with_pause

コード
! ./run.sh \
    --stage 6 \
    --g2p pyopenjtalk_accent_with_pause \
    --min_wav_duration 0.38 \
    --fs 44100 \
    --n_fft 2048 \
    --n_shift 512 \
    --dumpdir dump/44k \
    --win_length null \
    --tts_task gan_tts \
    --feats_extract linear_spectrogram \
    --feats_normalize none \
    --train_config ./conf/tuning/finetune_full_band_vits.yaml \
    --train_args "--init_param downloads/f3698edf589206588f58f5ec837fa516/exp/tts_train_vits_raw_phn_jaconv_pyopenjtalk_accent_with_pause/train.total_count.ave_10best.pth:tts:tts" \
    --tag finetune_full_band_vits_raw_phn_jaconv_pyopenjtalk_accent_with_pause

エラー
RuntimeError: Error(s) in loading state_dict for VITS:
	size mismatch for generator.decoder.output_conv.1.weight_v: copying a param with shape torch.Size([1, 32, 7]) from checkpoint, the shape in current model is torch.Size([1, 16, 7]).
	size mismatch for mel_loss.wav_to_mel.logmel.melmat: copying a param with shape torch.Size([513, 80]) from checkpoint, the shape in current model is torch.Size([1025, 80]).

https://github.com/espnet/espnet/tree/master/egs2/jvs/tts1#3-replace-token-list-with-pretrained-models-one-1
3. Replace token list with pretrained model's oneの手順は踏んだはずなのですが、解決しませんでした。

こちらのgoogle colabの公開されているノートブックを使用しました。
https://colab.research.google.com/drive/12buuborkXNFoef1UptjPr0jyw9qLc12u#scrollTo=8u__S68bvByu

**Comment by kan-bayashi** [comment, marked as answer]
こんにちは、ESPnetをご利用いただきありがとうございます。
ノートブックの詳細は存じ上げませんが、コンフィグで指定しているモデルと事前学習モデルが異なるのが原因です。
コンフィグではfull bandモデルになっており、事前学習モデルは通常の22khzモデルが使われているため、パラメータサイズ不一致が起きています。
どちらかを変更し、同じ構造のモデルを使うようにしてみてください。

**Reply by aoi-takumi** [reply]
ご返信ありがとうございます。遅くなり申し訳ございません。
こちらも初歩的な質問で大変申し訳ございません。
stage6実行時に今度は以下のエラーが発生しました。こちらは単純にcudaとpytorchが互換性のないバージョンになっているためでしょうか?
使用しているバージョンはそれぞれ以下の通りです。
pytorch : 1.8.2+cu111
CUDA    :  Cuda compilation tools, release 10.1, V10.1.243

**Reply by kan-bayashi** [reply]
エラーログを貼っていただけますか？

**Reply by aoi-takumi** [reply]
ご返信いただきありがとうございます。
すみません、肝心なエラーを貼り忘れておりました・・・
`RuntimeError: CUDA error: unknown error`

**Reply by kan-bayashi** [reply]
おそらくcudaのバージョンを不一致でしょう
例えばnvidia-smiしたときにcuda-11.1に対応しているか確認してみるといいかもしれません

**Reply by aoi-takumi** [reply]
ご回答ありがとうございます。
nvidia-smiで対応バージョンの確認ができるのですね。確認します。

**Reply by aoi-takumi** [reply]
遅くなりましたが、cudaのバージョン不一致のエラーは解消しました。
再度実行してみると、stage6実行時に再び「size mismatch for～」エラーが発生してしまいました。
事前学習モデルは通常の22khzモデルが使われているというお話をいただきましたが、事前学習モデルの設定をしている部分がどこかわかません。どちらの処理で行っているのでしょうか・・？

**Reply by kan-bayashi** [reply]
> 遅くなりましたが、cudaのバージョン不一致のエラーは解消しました。

こちらどのような手順を踏むことで解消したかを共有していただけませんか？
(同様の質問が出たときにとても助かります)

> 事前学習モデルは通常の22khzモデルが使われているというお話をいただきましたが、事前学習モデルの設定をしている部分がどこかわかません。

```
--train_args "--init_param downloads/f3698edf589206588f58f5ec837fa516/exp/tts_train_vits_raw_phn_jaconv_pyopenjtalk_accent_with_pause/train.total_count.ave_10best.pth:tts:tts"
```

の部分です。`tts_train_vits` は 22.05khzモデルで `tts_train_full_band_vits`が44.1khzモデルになります。
